# Video Game Recomendation System

## Business understanding

## Data understanding

In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

Dataset from kaggle on [here](https://www.kaggle.com/datasets/cynthiarempel/amazon-us-customer-reviews-dataset?select=amazon_reviews_us_Digital_Video_Games_v1_00.tsv)

In [24]:
df_raw = pd.read_csv('./data/Digital_Video_Games.tsv', sep='\t', header=0, error_bad_lines=False)

### Columns
- **marketplace**: 2 letter country code of the marketplace where the review was written.
- **customer_id**: Random identifier that can be used to aggregate reviews written by a single author.
- **review_id**: The unique ID of the review.
- **productid**: The unique Product ID the review pertains to. In the multilingual dataset the reviews for the same product in different countries can be grouped by the same productid.
- **product_parent**: Random identifier that can be used to aggregate reviews for the same product.
- **product_title**: Title of the product.
- **product_category**: Broad product category that can be used to group reviews (also used to group the dataset into coherent parts).
- **star_rating**: The 1-5 star rating of the review.
- **helpful_votes**: Number of helpful votes.
- **total_votes**: Number of total votes the review received.
- **vine**: Review was written as part of the Vine program.
- **verified_purchase**: The review is on a verified purchase.
- **review_headline**: The title of the review.
- **review_body**: The review text.
- **review_date**: The date the review was written.

In [26]:
df_raw.head()

marketplace  customer_id       review_id  product_id  product_parent  \
0          US     21269168   RSH1OZ87OYK92  B013PURRZW       603406193   
1          US       133437  R1WFOQ3N9BO65I  B00F4CEHNK       341969535   
2          US     45765011   R3YOOS71KM5M9  B00DNHLFQA       951665344   
3          US       113118  R3R14UATT3OUFU  B004RMK5QG       395682204   
4          US     22151364   RV2W9SGDNQA2C  B00G9BNLQE       640460561   

                                       product_title     product_category  \
0              Madden NFL 16 - Xbox One Digital Code  Digital_Video_Games   
1                                Xbox Live Gift Card  Digital_Video_Games   
2  Command & Conquer The Ultimate Collection [Ins...  Digital_Video_Games   
3                      Playstation Plus Subscription  Digital_Video_Games   
4  Saints Row IV - Enter The Dominatrix [Online G...  Digital_Video_Games   

   star_rating  helpful_votes  total_votes vine verified_purchase  \
0            2              2            3    N                 N   
1            5              0            0    N                 Y   
2            5              0            0    N                 Y   
3            5              0            0    N                 Y   
4            5              0            0    N                 Y   

                        review_headline  \
0  A slight improvement from last year.   
1                            Five Stars   
2               Hail to the great Yuri!   
3                            Five Stars   
4                            Five Stars   

                                         review_body review_date  
0  I keep buying madden every year hoping they ge...  2015-08-31  
1                                            Awesome  2015-08-31  
2  If you are prepping for the end of the world t...  2015-08-31  
3                                            Perfect  2015-08-31  
4                                           Awesome!  2015-08-31

In [21]:
df_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 144724 entries, 0 to 144723
Data columns (total 15 columns):
 #   Column             Non-Null Count   Dtype 
---  ------             --------------   ----- 
 0   marketplace        144724 non-null  object
 1   customer_id        144724 non-null  int64 
 2   review_id          144724 non-null  object
 3   product_id         144724 non-null  object
 4   product_parent     144724 non-null  int64 
 5   product_title      144724 non-null  object
 6   product_category   144724 non-null  object
 7   star_rating        144724 non-null  int64 
 8   helpful_votes      144724 non-null  int64 
 9   total_votes        144724 non-null  int64 
 10  vine               144724 non-null  object
 11  verified_purchase  144724 non-null  object
 12  review_headline    144722 non-null  object
 13  review_body        144722 non-null  object
 14  review_date        144721 non-null  object
dtypes: int64(5), object(10)
memory usage: 16.6+ MB


Some null values in the last 3 columns.

In [28]:
df_raw['marketplace'].unique()

array(['US'], dtype=object)

All reviews in dataset are from US customers

In [23]:
df_raw['customer_id'].nunique()

112891

112891 different customers 

In [29]:
df_raw['review_id'].nunique()

144724

Seems like there are no duplicate reviews in the set

In [22]:
df_raw['product_id'].nunique()

7939

7939 different products, but not all video games, found "Playstation Plus Subscription" in row 3, so all digitally sold products related video games, like subscriptions, seem to be included in the data set, will further look for other kind of product other than games and drop those.

In [31]:
df_raw['product_parent'].nunique()

7755

- **product_parent**: Random identifier that can be used to aggregate reviews for the same product.

seems like just another id but we have about 200 less data points in this column than product id

In [32]:
df_raw['product_title'].nunique()

6946

if i have 6946 different titles, i would expect the same amount of product_id, wonder if by droping everything that is not a game, like substriptions, would fix this discrepancy.

In [39]:
df_raw.iloc[3]['product_title']

'Playstation Plus Subscription'

it doesnt specify if the subscription is for 1, 3 or 12 months

## Data Preparation

## Modeling

## Evaluation

## Deployment